File: Alcoholtracker/Views/Home/DrinkEditSheet.swift
import SwiftUI

// MARK: - DrinkEditSheet
// Tap any logged drink in the history to edit its volume and time, or delete it.

struct DrinkEditSheet: View {

    let drink: Drink
    let profile: UserProfile?
    let onSave: (Double, Date) -> Void
    let onDelete: () -> Void

    @Environment(\.dismiss) private var dismiss
    @State private var volumeText: String
    @State private var timestamp: Date
    @State private var durationMinutes: Double
    @State private var showDeleteConfirm = false

    init(
        drink: Drink,
        profile: UserProfile?,
        onSave: @escaping (Double, Date) -> Void,
        onDelete: @escaping () -> Void
    ) {
        self.drink = drink
        self.profile = profile
        self.onSave = onSave
        self.onDelete = onDelete
        self._volumeText = State(initialValue: "\(Int(drink.volume))")
        self._timestamp = State(initialValue: drink.timestamp)
        self._durationMinutes = State(initialValue: drink.drinkDurationMinutes)
    }

    private var volume: Double {
        Double(volumeText.replacingOccurrences(of: ",", with: ".")) ?? 0
    }

    private var isValid: Bool { volume > 0 && volume <= 3000 }

    private var bacContribution: Double? {
        guard let p = profile, volume > 0 else { return nil }
        return BACCalculator.projectedPeak(
            volume: volume, abv: drink.abv, category: drink.category,
            profile: p, stomachStatus: p.defaultStomachStatus,
            drinkDurationMinutes: durationMinutes, conservative: p.conservativeForApp
        )
    }

    var body: some View {
        ZStack {
            Color.appBackground.ignoresSafeArea()
            VStack(spacing: 0) {
                RoundedRectangle(cornerRadius: 2.5)
                    .fill(Color.appBorder)
                    .frame(width: 36, height: 5)
                    .padding(.top, 12)
                    .padding(.bottom, 4)

                HStack(spacing: 14) {
                    DrinkIconView(drink: drink, size: 18)
                        .font(.system(size: 18, weight: .medium))
                        .foregroundStyle(Color.appAccent)
                        .frame(width: 40, height: 40)
                        .background(Color.appAccent.opacity(0.12))
                        .clipShape(RoundedRectangle(cornerRadius: 11))
                    VStack(alignment: .leading, spacing: 2) {
                        Text(drink.name)
                            .font(.appBodyBold)
                            .foregroundStyle(Color.appText)
                        Text("\(String(format: "%.1f", drink.abv)) % Alk.")
                            .font(.appCaption)
                            .foregroundStyle(Color.appTextDim)
                        if drink.mixerVolume > 0 {
                            Text("Spirituose \(Int(drink.volume - drink.mixerVolume)) ml, Mixer \(Int(drink.mixerVolume)) ml")
                                .font(.appMicro)
                                .foregroundStyle(Color.appTextMuted)
                        }
                    }
                    Spacer()
                    Button { dismiss() } label: {
                        Image(systemName: "xmark")
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundStyle(Color.appTextDim)
                            .frame(width: 32, height: 32)
                            .background(Color.appCard)
                            .clipShape(Circle())
                            .overlay(Circle().strokeBorder(Color.appBorder, lineWidth: 0.5))
                    }
                    .buttonStyle(.plain)
                }
                .padding(.horizontal, 20)
                .padding(.top, 12)
                .padding(.bottom, 20)

                ScrollView(showsIndicators: false) {
                    VStack(spacing: 16) {
                        VStack(alignment: .leading, spacing: 8) {
                            SectionLabel(text: "MENGE")
                            HStack(spacing: 8) {
                                TextField("ml", text: $volumeText)
                                    .keyboardType(.numberPad)
                                    .font(.appBody)
                                    .foregroundStyle(Color.appText)
                                    .autocorrectionDisabled()
                                Text("ml")
                                    .font(.appBodyBold)
                                    .foregroundStyle(Color.appTextDim)
                            }
                            .padding(.horizontal, 14)
                            .padding(.vertical, 12)
                            .background(Color.appCard)
                            .clipShape(RoundedRectangle(cornerRadius: 12))
                            .overlay(
                                RoundedRectangle(cornerRadius: 12)
                                    .strokeBorder(Color.appBorder, lineWidth: 0.5)
                            )
                        }

                        if drink.mixerVolume > 0 {
                            DESDrinkMixInfo(drink: drink)
                        }

                        VStack(alignment: .leading, spacing: 8) {
                            SectionLabel(text: "TRINKDAUER")
                            DurationChipRow(durationMinutes: $durationMinutes)
                            Text("Über welchen Zeitraum getrunken. Längere Dauer verteilt die Aufnahme und senkt den Peak.")
                                .font(.appMicro)
                                .foregroundStyle(Color.appTextMuted)
                        }

                        VStack(alignment: .leading, spacing: 8) {
                            SectionLabel(text: "UHRZEIT")
                            DatePicker("", selection: $timestamp, displayedComponents: [.hourAndMinute])
                                .datePickerStyle(.wheel)
                                .labelsHidden()
                                .frame(maxWidth: .infinity)
                                .background(Color.appCard)
                                .clipShape(RoundedRectangle(cornerRadius: 12))
                                .overlay(
                                    RoundedRectangle(cornerRadius: 12)
                                        .strokeBorder(Color.appBorder, lineWidth: 0.5)
                                )
                        }

                        if let bac = bacContribution {
                            HStack(spacing: 12) {
                                Image(systemName: "gauge.with.dots.needle.33percent")
                                    .font(.system(size: 16, weight: .medium))
                                    .foregroundStyle(Color.appAccent)
                                    .frame(width: 32, height: 32)
                                    .background(Color.appAccent.opacity(0.10))
                                    .clipShape(RoundedRectangle(cornerRadius: 9))
                                Text("Geschätzte Wirkung")
                                    .font(.appCaption)
                                    .foregroundStyle(Color.appTextDim)
                                Spacer()
                                Text(bac.signedPermilleString)
                                    .font(.appCaptionBold)
                                    .foregroundStyle(Color.statusOrange)
                            }
                            .padding(14)
                            .background(Color.appCard)
                            .clipShape(RoundedRectangle(cornerRadius: 12))
                            .overlay(
                                RoundedRectangle(cornerRadius: 12)
                                    .strokeBorder(Color.appBorder, lineWidth: 0.5)
                            )
                        }

                        PrimaryButton(title: "Speichern", icon: "checkmark", isDisabled: !isValid) {
                            // Persisted via the same context save onSave triggers.
                            drink.drinkDurationMinutes = durationMinutes
                            onSave(volume, timestamp)
                            dismiss()
                        }

                        Button { showDeleteConfirm = true } label: {
                            HStack(spacing: 8) {
                                Image(systemName: "trash")
                                    .font(.system(size: 14, weight: .medium))
                                Text("Drink entfernen")
                                    .font(.appBodyBold)
                            }
                            .foregroundStyle(Color.statusRed)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 14)
                            .background(Color.statusRed.opacity(0.10))
                            .clipShape(RoundedRectangle(cornerRadius: 16))
                            .overlay(
                                RoundedRectangle(cornerRadius: 16)
                                    .strokeBorder(Color.statusRed.opacity(0.3), lineWidth: 0.5)
                            )
                        }
                        .buttonStyle(.plain)
                    }
                    .padding(.horizontal, 20)
                    .padding(.bottom, 32)
                }
            }
        }
        .presentationDetents([.large])
        .presentationDragIndicator(.hidden)
        .confirmationDialog("Drink wirklich entfernen?", isPresented: $showDeleteConfirm, titleVisibility: .visible) {
            Button("Entfernen", role: .destructive) {
                onDelete()
                dismiss()
            }
            Button("Abbrechen", role: .cancel) {}
        }
    }
}

// MARK: - Mix details card

private struct DESDrinkMixInfo: View {
    let drink: Drink

    private var spiritML: Int   { Int(drink.volume - drink.mixerVolume) }
    private var mixerML: Int    { Int(drink.mixerVolume) }
    private var spiritPct: Int  { drink.volume > 0 ? Int((drink.volume - drink.mixerVolume) / drink.volume * 100) : 0 }

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            SectionLabel(text: "MIX-DETAILS")
            HStack(spacing: 0) {
                DESStat(label: "Spirituose", value: "\(spiritML) ml", color: .appAccent)
                Rectangle()
                    .fill(Color.appBorder)
                    .frame(width: 0.5, height: 32)
                DESStat(label: "Mixer", value: "\(mixerML) ml", color: .appTextDim)
                Rectangle()
                    .fill(Color.appBorder)
                    .frame(width: 0.5, height: 32)
                DESStat(label: "Spi.-Anteil", value: "\(spiritPct) %", color: .appAccent)
            }
            .padding(14)
            .background(Color.appCard)
            .clipShape(RoundedRectangle(cornerRadius: 12))
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .strokeBorder(Color.appBorder, lineWidth: 0.5)
            )
        }
    }
}

private struct DESStat: View {
    let label: String
    let value: String
    let color: Color

    var body: some View {
        VStack(spacing: 3) {
            Text(value)
                .font(.appCaptionBold)
                .foregroundStyle(color)
            Text(label)
                .font(.appMicro)
                .foregroundStyle(Color.appTextMuted)
        }
        .frame(maxWidth: .infinity)
    }
}

File: android/app/src/main/kotlin/de/tipau/promille/ui/screens/home/DrinkEditSheet.kt
package de.tipau.promille.ui.screens.home

import android.app.TimePickerDialog
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import de.tipau.promille.AppColors
import de.tipau.promille.bac.*
import de.tipau.promille.data.UserProfileEntity
import de.tipau.promille.repository.UserProfileRepository
import de.tipau.promille.ui.components.DrinkIconView
import de.tipau.promille.ui.components.DurationChipRow
import de.tipau.promille.ui.components.PrimaryButton
import de.tipau.promille.ui.components.SectionLabel
import java.time.Instant
import java.time.ZoneId
import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter
import java.util.Calendar
import java.util.Locale
import kotlin.math.roundToInt

/**
 * 1:1 Port of DrinkEditSheet.swift.
 * Full drink editing sheet with volume validation, mix details, timestamp edit,
 * live BAC impact preview, duration chip row, and delete confirmation.
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun DrinkEditSheet(
    drink: Drink,
    profileEntity: UserProfileEntity? = null,
    onDismiss: () -> Unit,
    onSave: (volume: Double, timestampSeconds: Long, durationMinutes: Double) -> Unit,
    onDuplicate: (() -> Unit)? = null,
    onFinishNow: (() -> Unit)? = null,
    onDelete: () -> Unit
) {
    val context = LocalContext.current
    var volumeText by remember { mutableStateOf(drink.volumeML.roundToInt().toString()) }
    var durationMinutes by remember { mutableStateOf(drink.drinkDurationMinutes.coerceAtLeast(0.0)) }
    var timestampSeconds by remember { mutableStateOf(drink.timestampEpochSeconds) }
    var showDeleteConfirm by remember { mutableStateOf(false) }

    val volume = volumeText.replace(',', '.').toDoubleOrNull() ?: 0.0
    val isValid = volume > 0.0 && volume <= 3000.0

    val profile = remember(profileEntity) {
        profileEntity?.let { UserProfileRepository.toProfile(it) } ?: Profile.DEFAULT
    }

    val autoDurationMinutes = remember(drink.category, volume) {
        DrinkDurationEstimator.estimate(drink.category, volume)
    }

    val effectiveDurationMinutes = remember(durationMinutes, autoDurationMinutes) {
        if (durationMinutes > 0.0) durationMinutes else autoDurationMinutes
    }

    val timeFormatter = remember {
        DateTimeFormatter.ofPattern("HH:mm", Locale.GERMAN).withZone(ZoneId.systemDefault())
    }

    val startTimeStr = remember(timestampSeconds) {
        timeFormatter.format(Instant.ofEpochSecond(timestampSeconds))
    }

    val estimatedEndTimeStr = remember(timestampSeconds, effectiveDurationMinutes) {
        val endSec = timestampSeconds + (effectiveDurationMinutes * 60).toLong()
        timeFormatter.format(Instant.ofEpochSecond(endSec))
    }

    val absorptionEndTimeStr = remember(drink.category, volume, durationMinutes, timestampSeconds, profile) {
        val windowMin = BacCalculator.absorptionWindowMinutes(
            category = drink.category,
            drinkDurationMinutes = if (effectiveDurationMinutes > 0.0) effectiveDurationMinutes else 15.0,
            gastricMinutes = profile.defaultStomachStatus.absorptionMinutes
        )
        val endSec = timestampSeconds + (windowMin * 60).toLong()
        timeFormatter.format(Instant.ofEpochSecond(endSec))
    }

    val bacContribution = remember(volume, drink.abv, drink.category, durationMinutes, profile) {
        if (!isValid) null
        else {
            val input = DrinkInput(
                offsetMinutes = 0.0,
                volumeML = volume,
                abv = drink.abv,
                category = drink.category,
                drinkDurationMinutes = if (effectiveDurationMinutes > 0.0) effectiveDurationMinutes else 15.0
            )
            BacCalculator.projectedPeak(
                drink = input,
                profile = profile,
                stomach = profile.defaultStomachStatus,
                conservative = profile.conservativeForApp
            )
        }
    }

    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)

    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState,
        containerColor = Color.Transparent,
        scrimColor = Color.Black.copy(alpha = 0.65f),
        dragHandle = null
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 16.dp)
                .padding(bottom = 24.dp, top = 16.dp)
                .clip(RoundedCornerShape(24.dp))
                .background(AppColors.background)
                .border(0.5.dp, AppColors.border, RoundedCornerShape(24.dp))
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp, vertical = 20.dp)
                    .verticalScroll(rememberScrollState()),
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                // Header
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Row(
                        verticalAlignment = Alignment.CenterVertically,
                        modifier = Modifier.weight(1f)
                    ) {
                        Box(
                            modifier = Modifier
                                .size(40.dp)
                                .clip(RoundedCornerShape(11.dp))
                                .background(AppColors.accent.copy(alpha = 0.12f)),
                            contentAlignment = Alignment.Center
                        ) {
                            DrinkIconView(
                                drink = drink,
                                tint = AppColors.accent,
                                size = 20.dp
                            )
                        }
                        Spacer(Modifier.width(14.dp))
                        Column {
                            Text(
                                // iOS: .appBodyBold (SemiBold, not Bold).
                                text = drink.name,
                                color = AppColors.text,
                                style = de.tipau.promille.AppText.bodyBold
                            )
                            Text(
                                text = "${String.format(Locale.GERMANY, "%.1f", drink.abv)} % Alk.",
                                color = AppColors.textDim,
                                style = de.tipau.promille.AppText.caption
                            )
                            if (drink.mixerVolumeML > 0.0) {
                                val spiritML = (drink.volumeML - drink.mixerVolumeML).roundToInt().coerceAtLeast(0)
                                val mixerML = drink.mixerVolumeML.roundToInt()
                                Text(
                                    text = "Spirituose $spiritML ml, Mixer $mixerML ml",
                                    color = AppColors.textMuted,
                                    style = de.tipau.promille.AppText.micro
                                )
                            }
                        }
                    }
                    de.tipau.promille.ui.components.AppIconCloseButton(onDismiss = onDismiss)
                }

                // Volume Section
                Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                    SectionLabel(text = "MENGE")
                    de.tipau.promille.ui.components.AppTextField(
                        value = volumeText,
                        onValueChange = { volumeText = it },
                        placeholder = "ml",
                        trailingIcon = { Text("ml", color = AppColors.textDim, fontSize = 14.sp, fontWeight = FontWeight.SemiBold) },
                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
                        singleLine = true,
                        modifier = Modifier.fillMaxWidth()
                    )
                }

                // Mix details card if cocktail/mixed
                if (drink.mixerVolumeML > 0.0) {
                    val spiritML = (drink.volumeML - drink.mixerVolumeML).roundToInt().coerceAtLeast(0)
                    val mixerML = drink.mixerVolumeML.roundToInt()
                    val spiritPct = if (drink.volumeML > 0) ((drink.volumeML - drink.mixerVolumeML) / drink.volumeML * 100).roundToInt() else 0

                    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                        SectionLabel(text = "MIX-DETAILS")
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .clip(RoundedCornerShape(12.dp))
                                .background(AppColors.card)
                                .border(0.5.dp, AppColors.border, RoundedCornerShape(12.dp))
                                .padding(14.dp),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            // iOS DESStat: value .appCaptionBold (SemiBold,
                            // not Bold), label .appMicro.
                            Column(horizontalAlignment = Alignment.CenterHorizontally, modifier = Modifier.weight(1f)) {
                                Text("$spiritML ml", color = AppColors.accent, style = de.tipau.promille.AppText.captionBold)
                                Text("Spirituose", color = AppColors.textMuted, style = de.tipau.promille.AppText.micro)
                            }
                            Box(modifier = Modifier.width(0.5.dp).height(32.dp).background(AppColors.border))
                            Column(horizontalAlignment = Alignment.CenterHorizontally, modifier = Modifier.weight(1f)) {
                                Text("$mixerML ml", color = AppColors.textDim, style = de.tipau.promille.AppText.captionBold)
                                Text("Mixer", color = AppColors.textMuted, style = de.tipau.promille.AppText.micro)
                            }
                            Box(modifier = Modifier.width(0.5.dp).height(32.dp).background(AppColors.border))
                            Column(horizontalAlignment = Alignment.CenterHorizontally, modifier = Modifier.weight(1f)) {
                                Text("$spiritPct %", color = AppColors.accent, style = de.tipau.promille.AppText.captionBold)
                                Text("Spi.-Anteil", color = AppColors.textMuted, style = de.tipau.promille.AppText.micro)
                            }
                        }
                    }
                }

                // Duration Section
                Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                    SectionLabel(text = "TRINKDAUER")
                    DurationChipRow(
                        durationMinutes = durationMinutes,
                        estimatedMinutes = autoDurationMinutes,
                        onDurationMinutesChange = { durationMinutes = it }
                    )
                    Text(
                        text = "Start $startTimeStr · fertig ca. $estimatedEndTimeStr",
                        color = AppColors.textMuted,
                        style = de.tipau.promille.AppText.micro
                    )
                    Text(
                        text = "Aufnahme ca. bis $absorptionEndTimeStr",
                        color = AppColors.textMuted,
                        style = de.tipau.promille.AppText.micro
                    )
                }

                // Timestamp Section
                Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                    SectionLabel(text = "UHRZEIT")
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(12.dp))
                            .background(AppColors.card)
                            .border(0.5.dp, AppColors.border, RoundedCornerShape(12.dp))
                            .clickable {
                                val cal = Calendar.getInstance().apply {
                                    timeInMillis = timestampSeconds * 1000
                                }
                                TimePickerDialog(
                                    context,
                                    { _, hourOfDay, minute ->
                                        val zdt = Instant.ofEpochSecond(timestampSeconds)
                                            .atZone(ZoneId.systemDefault())
                                            .withHour(hourOfDay)
                                            .withMinute(minute)
                                        timestampSeconds = zdt.toEpochSecond()
                                    },
                                    cal.get(Calendar.HOUR_OF_DAY),
                                    cal.get(Calendar.MINUTE),
                                    true
                                ).show()
                            }
                            .padding(horizontal = 14.dp, vertical = 14.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        // iOS: .appBody / .appBodyBold (SemiBold, not Bold).
                        Text("Startzeit", color = AppColors.textDim, style = de.tipau.promille.AppText.body)
                        Text(startTimeStr, color = AppColors.text, style = de.tipau.promille.AppText.bodyBold)
                    }
                }

                // Estimated BAC Contribution Preview
                if (bacContribution != null && bacContribution > 0.0) {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(12.dp))
                            .background(AppColors.card)
                            .border(0.5.dp, AppColors.border, RoundedCornerShape(12.dp))
                            .padding(14.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text("Geschätzte Wirkung", color = AppColors.textDim, style = de.tipau.promille.AppText.caption, modifier = Modifier.weight(1f))
                        Text(
                            // iOS: .appCaptionBold (SemiBold, not Bold).
                            text = "+${String.format(Locale.GERMANY, "%.2f", bacContribution)} ‰",
                            color = AppColors.statusOrange,
                            style = de.tipau.promille.AppText.captionBold
                        )
                    }
                }

                // Save Button
                PrimaryButton(
                    text = "Speichern",
                    enabled = isValid,
                    onClick = {
                        onSave(volume, timestampSeconds, durationMinutes)
                        onDismiss()
                    }
                )

                // Delete Button
                Button(
                    onClick = { showDeleteConfirm = true },
                    colors = ButtonDefaults.buttonColors(
                        containerColor = AppColors.statusRed.copy(alpha = 0.10f),
                        contentColor = AppColors.statusRed
                    ),
                    shape = RoundedCornerShape(16.dp),
                    modifier = Modifier
                        .fillMaxWidth()
                        .border(0.5.dp, AppColors.statusRed.copy(alpha = 0.3f), RoundedCornerShape(16.dp))
                        .padding(vertical = 2.dp)
                ) {
                    Row(
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        modifier = Modifier.padding(vertical = 6.dp)
                    ) {
                        Icon(Icons.Filled.Delete, null, modifier = Modifier.size(16.dp))
                        // iOS: .appBodyBold (17sp SemiBold) - was 16sp Bold.
                        Text("Drink entfernen", style = de.tipau.promille.AppText.bodyBold)
                    }
                }
            }
        }
    }

    if (showDeleteConfirm) {
        AlertDialog(
            onDismissRequest = { showDeleteConfirm = false },
            title = { Text("Drink wirklich entfernen?", color = AppColors.text, fontWeight = FontWeight.Bold) },
            text = { Text("Dieser Eintrag wird dauerhaft aus deiner Session und der Historie gelöscht.", color = AppColors.textDim) },
            confirmButton = {
                TextButton(
                    onClick = {
                        showDeleteConfirm = false
                        onDelete()
                        onDismiss()
                    }
                ) {
                    Text("Entfernen", color = AppColors.statusRed, fontWeight = FontWeight.Bold)
                }
            },
            dismissButton = {
                TextButton(onClick = { showDeleteConfirm = false }) {
                    Text("Abbrechen", color = AppColors.textDim)
                }
            },
            containerColor = AppColors.card,
            shape = RoundedCornerShape(20.dp)
        )
    }
}

